### ISSUE TITLE
Remove shipping code from runtime-assets

### ISSUE THREAD

**Issue by eerhardt** [dotnet/runtime#60273]
https://github.com/dotnet/runtime-assets/pull/160 added some Mono.Cecil code to https://github.com/dotnet/runtime-assets that our shipping product takes a dependency on (through a "Sources" NuGet package):

https://github.com/dotnet/runtime/blob/c028f39d6ca2f8d26cd6d9351a4d4eb308a18d23/src/tasks/MonoTargetsTasks/ILStrip/AssemblyStripper/AssemblyStripper.csproj#L16

We shouldn't have shipping code in https://github.com/dotnet/runtime-assets. That repo doesn't have a servicing story, and so when we need to service it, it is going to be a pain. That repo is only for large binary files that we don't want checked into dotnet/runtime - like test data. We have copied other "3rd party" source code in dotnet/runtime for other scenarios (e.g. https://github.com/dotnet/runtime/tree/main/src/coreclr/pal/src/libunwind). We should follow the same pattern here for this Mono.Cecil code (until it is re-written and removed according to the README added in https://github.com/dotnet/runtime-assets/pull/160).

cc @ericstj @ViktorHofer @steveisok @akoeplinger

**Comment by eerhardt** [comment]
> Maybe we just need runtime-assets to opt into source-build?

My approach to fixing this in source-build is to just not build `ILStrip.csproj` during source-build. It isn't necessary to build during source-build because it doesn't get included in the official SDK installer. It only is included in https://github.com/dotnet/runtime/blob/d262c21063bbb5b385a776e4ce24f7132eb918dc/src/mono/nuget/Microsoft.NET.Runtime.MonoTargets.Sdk/Microsoft.NET.Runtime.MonoTargets.Sdk.pkgproj#L9, which doesn't get built already.

**Comment by ViktorHofer** [comment]
> I see this repo has an internal mirror and presumably that's where the builds happen and publish. If it's following all the normal rules for a product repo maybe its ok.

Correct, the repository does everything necessary to be part of the dependency flow. We already had product dependencies present during 5.0 and therefore have a release/5.0 servicing branch.

> My approach to fixing this in source-build is to just not build ILStrip.csproj during source-build.

Agreed that this makes most sense. You should be able achieve that by just setting a property in the project file: https://github.com/dotnet/arcade/blob/57ac10418e39271319394a230f7e3a8f4ef9f15b/src/Microsoft.DotNet.ApiCompat/src/Microsoft.DotNet.ApiCompat.csproj#L6.

**Comment by steveisok** [comment]
I marked this with a 7.0 milestone, but the chances are good this slips beyond 7.

**Comment by ViktorHofer** [comment]
Kindly asking for an update on this @akoeplinger @steveisok. Were we able to remove the AssemblyStripper code base from runtime-assets?

**Comment by steveisok** [comment]
No, we haven't been able to remove it.  I think it's going to take some time to adequately replace what we have and I think we may need some help in doing so in the 8.0 timeframe.
